### PR TITLE
[shopsys] switch domain default currencies

### LINKED ISSUE #305: Demo data - currencies on domains
### What is happening

The prices on the first (eng) domain are in CZK currency and prices on the second (czech) domain are in EUR currency.

### Expected result

There should be CZk currency on the second domain and EUR currency on the first domain

### PR DESCRIPTION
**Pull request by vitek-rostislav**
- all domains now use EUR
- the only exception is second domain that uses CZK

| Q             | A
| ------------- | ---
|Description, reason for the PR| The currenct state is really weird - Czech domain uses EUR and English domain uses CZK currency - we have always wanted to fix this
|New feature| No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| closes #305, closes #1317
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

TODO:
- [ ] ~product (and probably other) input prices should be altered - now they are too high (or did I something wrong? :thinking:)~
- [ ] ~there should be a warning in upgrade notes - the migration should be skipped~

### CODE REVIEW

File: project-base/src/Shopsys/ShopBundle/Migrations/Version20190918155540.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+{
+    /**
+     * @param \Doctrine\DBAL\Schema\Schema $schema
+     */
+    public function up(Schema $schema)
+    {
+        $this->sql('UPDATE currencies SET exchange_rate = 0.04 WHERE code = \'CZK\'');
+        $this->sql('INSERT INTO currencies (id, name, code, exchange_rate) VALUES (2, \'Euro\', \'EUR\', 1)');
+        $this->sql('UPDATE setting_values SET value = 2 WHERE name = \'defaultCurrencyId\'');
+        $this->sql('UPDATE setting_values SET value = 2 WHERE name = \'defaultDomainCurrencyId\' AND domain_id = 1');
```
Comment (pk16011990) [inline project-base/src/Shopsys/ShopBundle/Migrations/Version20190918155540.php]: This migration is OK on project-base. However it is problem for existing projects. 

Why not modify existing migrations?

Yes, I agree with "Migrations can not be modified", but this is nice example for it. In my opinion it does not have any disadvantage :grin: in this case.

Yes, projects can skip this migration. But why, If we can prevent it.

Comment (vitek-rostislav) [reply project-base/src/Shopsys/ShopBundle/Migrations/Version20190918155540.php]: :thinking: I think that I am ok with changing the existing migration... :slightly_smiling_face:

**Comment by vitek-rostislav** [comment]
Hi @TomasLudvik...almost a month ago, you wanted me to create this PR..however, there is no reaction so far, why? :disappointed:

File: project-base/src/Shopsys/ShopBundle/DataFixtures/Demo/CurrencyDataFixture.php [hunk 2]
```diff
@@ -62,15 +63,7 @@ public function load(ObjectManager $manager)
...
-            $currencyData = $this->currencyDataFactory->create();
-            $currencyData->name = 'Euro';
-            $currencyData->code = Currency::CODE_EUR;
-            $currencyData->exchangeRate = '25';
-            $currencyData->minFractionDigits = Currency::DEFAULT_MIN_FRACTION_DIGITS;
-            $currencyData->roundingType = Currency::ROUNDING_TYPE_HUNDREDTHS;
-            $currencyEuro = $this->currencyFacade->create($currencyData);
-            $this->addReference(self::CURRENCY_EUR, $currencyEuro);
-        }
+        $currencyCzk = $this->currencyFacade->getById(2);
```
Comment (TomasLudvik) [inline project-base/src/Shopsys/ShopBundle/DataFixtures/Demo/CurrencyDataFixture.php]: You should rename this to `$currencyEur`

File: project-base/src/Shopsys/ShopBundle/Migrations/Version20190918155540.php (hunk 1 above)
Comment (TomasLudvik) [reply project-base/src/Shopsys/ShopBundle/Migrations/Version20190918155540.php]: @pk16011990 has a valid point, can you please update it @vitek-rostislav ?

File: project-base/src/Shopsys/ShopBundle/Migrations/Version20190918155540.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ * This migration changes the default currencies settings - it should be ran on the clean project-base installation only!
+ */
+class Version20190918155540 extends AbstractMigration
+{
+    /**
+     * @param \Doctrine\DBAL\Schema\Schema $schema
+     */
+    public function up(Schema $schema)
+    {
+        $this->sql('UPDATE currencies SET exchange_rate = 0.04 WHERE code = \'CZK\'');
```
Comment (TomasLudvik) [inline project-base/src/Shopsys/ShopBundle/Migrations/Version20190918155540.php]: I think, that exchange rate for CZK should stay 1 and exchange rate for EUR should be 25

File: project-base/src/Shopsys/ShopBundle/Migrations/Version20190918155540.php (hunk 1 above)
Comment (TomasLudvik) [inline project-base/src/Shopsys/ShopBundle/Migrations/Version20190918155540.php]:
```suggestion
        $this->sql('UPDATE setting_values SET value = 2 WHERE name = \'defaultDomainCurrencyId\' AND domain_id != 2');
```

File: project-base/src/Shopsys/ShopBundle/Migrations/Version20190918155540.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        $this->sql('INSERT INTO currencies (id, name, code, exchange_rate) VALUES (2, \'Euro\', \'EUR\', 1)');
```
Comment (TomasLudvik) [inline project-base/src/Shopsys/ShopBundle/Migrations/Version20190918155540.php]: currencies now require min_fraction_digits and rounding_type

**Comment by pk16011990** [comment]
Hi, it closed my PR and this PR is stucked...?!

**Comment by vitek-rostislav** [comment]
Hi, I'll try to address the objections and rebase my branch ASAP

**Pull request merged**